### DISCUSSION TITLE
Can not invalidate cache tags

### DISCUSSION THREAD

**Discussion by alpzaf** [reduxjs/redux-toolkit#4692] [Q&A] [answered]
When I'm using injectEndpoints api the RTK does not recognize tag types. I shared below the screenshots of my setup
<img width="653" alt="Screenshot 2024-11-05 at 12 03 23" src="https://github.com/user-attachments/assets/b8949ddf-d77c-419f-ba42-468d452449ba">
<img width="753" alt="Screenshot 2024-11-05 at 12 04 01" src="https://github.com/user-attachments/assets/a111acbb-6282-48cb-85e7-ccafc3de644f">
<img width="937" alt="Screenshot 2024-11-05 at 12 04 47" src="https://github.com/user-attachments/assets/80805883-6371-4a99-a311-8e9820c6d78a">
<img width="713" alt="Screenshot 2024-11-05 at 12 05 28" src="https://github.com/user-attachments/assets/74fe1c3e-c265-4d1c-bda8-a4a15dcbbe14">
<img width="599" alt="Screenshot 2024-11-05 at 12 05 49" src="https://github.com/user-attachments/assets/53e577f9-c435-48af-80b0-e94a0405a1f0">
<img width="770" alt="Screenshot 2024-11-05 at 12 06 52" src="https://github.com/user-attachments/assets/374db854-6bc3-46f6-bde7-d9d8a4d987af">
<img width="431" alt="Screenshot 2024-11-05 at 12 07 37" src="https://github.com/user-attachments/assets/c98c1901-ad8b-4044-8176-2923475530cc">
<img width="1006" alt="image" src="https://github.com/user-attachments/assets/ebf93d79-39c3-4582-8db0-9d1c0738cfc9">

**Comment by markerikson** [comment]
Can you clarify what you mean by "doesn't recognize tag types"?

**Reply by alpzaf** [reply]
At first, I thought even I specified my tag types caches weren't invalidating then after I tried to debug and log out tags in my injected endpoints I couldn't see any of my tag types. After I realized can't see tags in redux dev tools too. Because of this I shared to last screenshot. Then I thought redux doesn't know which tags I specified.

**Reply by alpzaf** [reply]
I have also tried to enhance endpoints api to addTagTypes but it doesn't work either

**Comment by phryneas** [comment]
I don't see `'people'` in your `tagTypes`

**Reply by alpzaf** [reply]
There is.

**Reply by phryneas** [reply]
Oh, blind. Just out of interest, does it work if you remove the persistance?

**Reply by alpzaf** [reply, marked as answer]
Actually, I'd spotted the issue right in the first place. The problem was that my injected endpoints couldn't get any of tagTypes because after I realized I put the provide and invalidate arrays in the wrong place they were in query object.


    **
    // PREVIOUS
    getAllPeople: build.query<ApiResponse<Person[]>, QueryParams | string>({
      query: (query) => ({
        url: '${SERVICE_URL}/people/filter?${buildQueryString(query as QueryParams)}',
        providesTags: ['People'] // -> wrong place. (does not throw any error)
      }),
    }),
    
    **


    **
    // WORKING ONE
    getAllPeople: build.query<ApiResponse<Person[]>, QueryParams | string>({
      query: (query) => ({
        url: '${SERVICE_URL}/people/filter?${buildQueryString(query as QueryParams)}'
      }),
      providesTags: ['People'] // -> right place.   (throws TagDescription<T> error when tags not matches with api's)
    }),
    
    **
    
So the issue was when I try to provide my tags into the query object it doesn't throw any error, this tricked me the whole time.
I solved the problem when I isolated the codebase in a different setup and saw TagDescription error.

**Reply by markerikson** [reply]
Huh, very surprised TS didn't catch that!

**Reply by EskiMojo14** [reply]
[No Excess Property Checks When Comparing Functions](https://www.totaltypescript.com/books/total-typescript-essentials/the-weird-parts#no-excess-property-checks-when-comparing-functions) @markerikson
